File: 09_Sorting_BubbleSort.py
"""
https://www.hackerrank.com/challenges/ctci-bubble-sort

"""

# Complete the countSwaps function below.
def countSwaps(a):
    s = 0
    for i in range(len(a)):
        for j in range(len(a) - 1):
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]
                s += 1
    print("Array is sorted in {} swaps.".format(s))
    print("First Element:", a[0])
    print("Last Element:", a[-1])

if __name__ == '__main__':
    n = int(input())
    a = list(map(int, input().rstrip().split()))
    countSwaps(a)

File: 09_Sorting_BubbleSort.cpp
//
// Created by jan on 29.12.19.
// https://www.hackerrank.com/challenges/ctci-bubble-sort
//

#include <bits/stdc++.h>

using namespace std;

vector<string> split_string(string);

// Complete the countSwaps function below.
void countSwaps(vector<int> a) {
    int s = 0;
    for (int i = 0; i < a.size(); i++){
        for (int j = 0; j < a.size() - 1; j++){
            if (a[j] > a[j + 1]){
                swap(a[j], a[j + 1]);
                s++;

            }
        }
    }
    cout << "Array is sorted in " << s << " swaps." << endl;
    cout << "First Element: " << a.front() << endl;
    cout << "Last Element: " << a.back() << endl;
}